features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetric.kt

```diff
@@ -72,6 +76,41 @@ internal class SessionEndedMetric(
         missedEventCountByType[missedEventType] = (missedEventCountByType[missedEventType] ?: 0) + 1
     }
 
+    fun onUploadQualityTracked(event: UploadQualityEvent) {
+        val track = uploadQuality.getOrPut(event.track) {
+            mutableMapOf<String, Any?>().apply {
```

Comment:
tip: Maybe it's better to have `mutableMapOf<UploadQualityMetric, Any?>()` ? then you'll need to call `.key` only when serializing it